cmd/apichecker/apicheck_test.go

```diff
@@ -0,0 +1,694 @@
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"reflect"
+	"testing"
+
+	"github.com/thrasher-corp/gocryptotrader/common/convert"
+	exchange "github.com/thrasher-corp/gocryptotrader/exchanges"
+	"github.com/thrasher-corp/gocryptotrader/log"
+)
+
+var (
+	testAPIKey      = ""
+	testAPIToken    = ""
+	testChecklistID = ""
+	testCardID      = ""
+	testListID      = ""
+	testBoardID     = ""
+	testBoardName   = ""
+	canTestMainFile = false
+)
+
+func TestMain(m *testing.M) {
+	setTestVars()
+	testMode = true
+	c := log.GenDefaultSettings()
+	c.Enabled = convert.BoolPtr(true)
+	log.GlobalLogConfig = &c
+	log.Infoln(log.Global, "set verbose to true for more detailed output")
+	var err error
+	configData, err = readFileData(jsonFile)
+	if err != nil {
+		log.Error(log.Global, err)
+		os.Exit(1)
+	}
+	testConfigData, err = readFileData(testJSONFile)
+	if err != nil {
+		log.Error(log.Global, err)
+		os.Exit(1)
+	}
+	usageData = testConfigData
+	setTestVars()
+	defer os.Exit(m.Run())
```

Comment:
Be careful with multiple defers as it is FILO.  In this context the function below will execute first and then it will run m.Run()